ImageList/Services/OAuth2TokenStorage.swift

```diff
@@ -7,19 +7,146 @@
 
 import Foundation
 
-final class OAuth2TokenStorage: OAuth2TokenStorageProtocol {
-    private let userDefaults = UserDefaults.standard
+protocol OAuth2TokenStorageProtocol {
+    var token: String? { get set }
+    func deleteToken()
+}
+
+final class OAuth2TokenStorage: OAuth2TokenStorageProtocol{
+    // MARK: - PUBLIC
+    public var token: String? {
+        get {
+            guard let data = get(
+                service: KeyPath.imageList.rawValue,
+                account: KeyPath.key.rawValue)
+            else {
+                print("OAuth2TokenStorage failed to read password")
+                return nil
+            }
+            return String(decoding: data, as: UTF8.self)
+        }
+        set(newValue) {
```

Comment:
Даже не знала что такой синтаксис существует, спасибо :) но давай локаничную форму:
```suggestion
        set {
```